cri/v1alpha1/cri.go

```diff
@@ -186,6 +192,12 @@ func (c *CriManager) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 	if err != nil {
 		return nil, fmt.Errorf("failed to create sandbox root directory: %v", err)
 	}
+	defer func() {
+		// If running sandbox failed, clean up the sandbox directory.
+		if retErr != nil {
+			os.RemoveAll(sandboxRootDir)
```

Comment:
How about declaring the variable `sandboxRootDir` before the first `defer func()`, and then I think we could combine the two `defer`s into a single one. WDYT? @YaoZengzeng

Reply:
Actually we can't, because `sandboxRootDir := path.Join(c.SandboxBaseDir, id)` and we should get the container ID first :)